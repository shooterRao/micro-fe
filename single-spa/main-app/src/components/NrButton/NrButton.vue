<!--
 * @Author: raojw
 * @Description: NrButton
 -->
<template>
  <button type="button" :class="buttonClass" :style="buttonStyle" @click.stop.prevent="handleClick">
    <template v-if="loading">
      <i class="icon-loading ivu-load-loop ivu-icon ivu-icon-ios-loading"></i>
    </template>
    <slot></slot>
  </button>
</template>

<script>
const prefixCls = "nr-button";

export default {
  name: "NrButton",
  props: {
    // 一级、二级、三级
    level: {
      type: Number,
      default: 1
    },
    // 类型
    // default, primary, back
    type: {
      type: String,
      default: "default"
    },
    // 是否禁用
    disabled: {
      type: Boolean,
      default: false
    },
    loading: {
      type: Boolean,
      default: false
    },
    // 背景色
    bgColor: {
      type: String
    },
    // 字体颜色
    fontColor: {
      type: String
    }
  },
  computed: {
    buttonClass() {
      const { level, type, disabled, loading } = this;
      return [
        `${prefixCls}`,
        `${prefixCls}-${level}`,
        `${prefixCls}-${type}`,
        disabled && "disabled",
        loading && "loading"
      ];
    },
    buttonStyle() {
      const { bgColor, fontColor } = this;
      const style = {};
      if (bgColor) {
        style.backgroundColor = bgColor;
        style.borderColor = bgColor;
      }
      if (fontColor) {
        style.color = fontColor;
      }
      return style;
    }
  },
  methods: {
    handleClick(event) {
      if (this.disabled || this.loading) {
        return;
      }
      // 返回 event 参数支持 @click.stop.prevent 写法
      this.$emit("click", event);
    }
  }
};
</script>

<style scoped lang="scss">
// 默认
$default-bg-color: #1890ff;
$default-font-color: #ffffff;

// 镂空效果
$primary-bg-color: #ffffff;
$primary-font-color: #1890ff;

// 退出(返回)按钮
$back-bg-color: #343c4c;
$back-font-color: #ffffff;

.nr-button {
  display: inline-block;
  text-align: center;
  touch-action: manipulation;
  line-height: 1;
  outline: 0;
  cursor: pointer;
  user-select: none;
  border-radius: 0.21rem;
  border: 1px solid transparent;
  transition: color 0.2s linear, background-color 0.2s linear,
    border 0.2s linear, box-shadow 0.2s linear;
  .icon-loading {
    margin-right: 4px;
    vertical-align: inherit;
  }
  &-1 {
    font-size: 0.83rem;
    padding: 0.66em 0.875em;
    .icon {
      font-size: 0.83rem;
    }
  }
  &-2 {
    font-size: 0.73rem;
    padding: 0.5em 1em;
    .icon {
      font-size: 0.73rem;
    }
  }
  &-3 {
    font-size: 0.73rem;
    padding: 0.3em 1em;
    .icon {
      font-size: 0.73rem;
    }
  }
  &-default {
    background-color: $default-bg-color;
    border-color: $default-bg-color;
    color: $default-font-color;
  }
  &-primary {
    background-color: $primary-bg-color;
    border-color: $primary-font-color;
    color: $primary-font-color;
    &:hover:not(.disabled) {
      color: #ffffff;
      border-color: #1890ff;
      background-color: #1890ff;
    }
  }
  &-back {
    background-color: $back-bg-color;
    border-color: $back-bg-color;
    color: $back-font-color;
  }
  &:hover:not(.nr-button-primary) {
    opacity: 0.8;
  }
  &.disabled {
    cursor: not-allowed;
    opacity: 0.4;
  }
  &.loading {
    opacity: 0.8;
    pointer-events: none;
  }
}
</style>
