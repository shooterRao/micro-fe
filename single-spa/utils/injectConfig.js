Promise.all([
  System.import('axios'),
  System.import('vuex'),
  System.import('micro-store')
]).then(function(modules) {
  const axios = modules[0];
  const vuex = modules[1];

  const microStore = modules[2];

  const store = new microStore.default();

  axios.get('/config/appConfig.json').then(res => {
    store.register('globalState', {
      state: {
        config: null,
        sysName: ''
      },
      mutations: {
        setConfig(state, data) {
          state.config = data;
        },
        setSysName(state, sysName) {
          state.sysName = sysName;
        }
      }
    });

    store.commit('globalState', 'setConfig', res.data);
    store.commit('globalState', 'setSysName', res.data.name);

    const moduleState = store.getState('globalState');

    console.log(moduleState.state);
  });
});
