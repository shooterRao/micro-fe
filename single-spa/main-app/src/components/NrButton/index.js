import NrButton from "./NrButton.vue";

export default NrButton;
