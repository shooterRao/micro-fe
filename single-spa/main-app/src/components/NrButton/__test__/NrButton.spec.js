import { mount } from "@vue/test-utils";
import NrButton from "@/components/NrButton";
import { asyncExpect } from "@/../tests/utils";

describe("NrButton", () => {
  const wrapper = mount({
    render() {
      return <NrButton>follow</NrButton>;
    }
  }, { sync: false });
  it("正确渲染组件标签", () => {
    // expect(wrapper.html()).toContain(`<button type="button" class="nr-button nr-button-1 nr-button-default">follow</button>`);
    expect(wrapper.html()).toMatchSnapshot();
  });
  describe(":props ", () => {
    it(":levels - 创建二级按钮", () => {
      const wrapper = mount(
        {
          render() {
            return <NrButton level={2}>按钮</NrButton>;
          }
        },
        { sync: false }
      );
      expect(wrapper.contains(".nr-button-2")).toBe(true);
    });
    it(":type - 创建 primary 按钮", () => {
      const wrapper = mount({
        render() {
          return <NrButton type="primary">按钮</NrButton>;
        }
      });
      expect(wrapper.contains(".nr-button-primary")).toBe(true);
    });
    it(":disabled - 创建 disabled 按钮", () => {
      const wrapper = mount(
        {
          render() {
            return <NrButton disabled={true}>按钮</NrButton>;
          }
        },
        { sync: false }
      );
      expect(wrapper.contains(".disabled")).toBe(true);
    });
    it(":loading - 创建 loading 按钮", () => {
      const wrapper = mount(
        {
          render() {
            return <NrButton loading={true}>按钮</NrButton>;
          }
        },
        { sync: false }
      );
      expect(wrapper.contains(".icon-loading")).toBe(true);
    });
    it(":fontColor 自定义按钮样式", () => {
      const wrapper = mount(
        {
          render() {
            return (
              <NrButton bgColor="#cccccc" fontColor="#ffffff">
                按钮
              </NrButton>
            );
          }
        },
        { sync: false }
      );
      expect(wrapper.element.style.backgroundColor).toBe("rgb(204, 204, 204)");
      expect(wrapper.element.style.color).toBe("rgb(255, 255, 255)");
    });
  });
  describe("@event", () => {
    it("单击按钮触发事件，改变 loading 状态", async () => {
      const DefaulButton = {
        data() {
          return {
            loading: false
          };
        },
        methods: {
          handleClickButton() {
            this.loading = true;
          }
        },
        render() {
          return (
            <NrButton loading={this.loading} onClick={this.handleClickButton} />
          );
        }
      };
      const wrapper = mount(DefaulButton, { sync: false });
      wrapper.trigger("click");
      await asyncExpect(() => {
        expect(wrapper.find(".icon-loading").exists()).toBe(true);
      });
    });
  });
});
