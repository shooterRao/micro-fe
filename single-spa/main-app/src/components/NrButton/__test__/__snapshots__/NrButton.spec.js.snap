// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NrButton 正确渲染组件标签 1`] = `
<button type="button" class="nr-button nr-button-1 nr-button-default">
  <!----> follow</button>
`;
